libnmstate/nm/team.py

```diff
@@ -17,14 +17,81 @@
 # along with this program. If not, see <https://www.gnu.org/licenses/>.
 #
 
-from libnmstate.error import NmstateNotImplementedError
+import json
+
+from libnmstate.nm import connection as nm_connection
+from libnmstate.nm import nmclient
+from libnmstate.schema import Interface
 from libnmstate.schema import Team
 
 
-def create_setting(iface_state):
-    setting = None
+def create_setting(iface_state, base_con_profile):
+    team_setting = None
     team_config = iface_state.get(Team.CONFIG_SUBTREE)
-    if team_config:
-        raise NmstateNotImplementedError
+    if base_con_profile:
+        if team_config:
+            team_setting = base_con_profile.get_setting_duplicate(
+                nmclient.NM.SETTING_TEAM_SETTING_NAME
+            )
+
+            if not team_setting:
+                team_setting = nmclient.NM.SettingTeam.new()
+
+            team_config = _convert_team_config_to_setting_format(
+                team_config, iface_state.get(Interface.NAME)
+            )
+
+            team_setting.props.config = json.dumps(team_config)
+
+    return team_setting
+
+
+def _convert_team_config_to_setting_format(team_config, ifname):
+    team_config["device"] = ifname
+
+    if team_config.get(Team.PORT_SUBTREE):
+        team_port = {
+            port.pop(Team.Port.NAME): port
+            for port in team_config[Team.PORT_SUBTREE]
+        }
+        team_config[Team.PORT_SUBTREE] = team_port
+
+    return team_config
+
+
+def get_info(device):
+    """
+    Provide the current active teamd values for an interface.
+    """
+    info = {}
+
+    connection = nm_connection.ConnectionProfile()
+    connection.import_by_device(device)
+    if not connection.profile:
+        return info
+
+    team_setting = connection.profile.get_setting_by_name(
+        nmclient.NM.SETTING_TEAM_SETTING_NAME
+    )
+
+    if team_setting:
+        team_json = team_setting.get_config()
+        if team_json:
+            team_config = json.loads(team_json)
+            team_config = _convert_team_config_to_config_format(team_config)
+            info[Team.CONFIG_SUBTREE] = team_config
+
+    return info
+
+
+def _convert_team_config_to_config_format(team_config):
```

Comment:
"config to config format"?

Reply:
Oops. This name is confusing again. I renamed it to `_convert_teamd_config_to_nmstate_config`. Thanks!